File: obsolete/scan/mysql_to_mongo.py
import MySQLdb
from pymongo import MongoClient
import json
import datetime

db = MySQLdb.connect(host="localhost", # your host, usually localhost
                     user="tagg", # your username
                      # passwd="megajonhy", # your password
                     db="arachne") # name of the data base

client = MongoClient('localhost', 27017)
mydb = client.mydb
mydb.tscans.drop()
collection = mydb.tscans


# you must create a Cursor object. It will let
#  you execute all the query you need
cur = db.cursor() 

# Use all the SQL you like
cur.execute("SELECT * FROM tscans;")

# print cur.description
# print all the first cell of all the rows

# three sub-collections:
event_id_tags = [ "recoVer", "det", "run", "subrun", "gate", "slice", "filename" ]
scan_id_tags = [ "user_name", "modification_date" ]
checksum_tags = [ "idhits", "odhits", "tmean", "calpe"]

adate = datetime.datetime.utcnow();

for row in cur.fetchall() :
    event_id = {}
    scan_id = {}
    checksum = {}
    scan = {}
    for icol,col in enumerate(cur.description) :
        colname = col[0]
        val = row[icol]
        if type(val) == type(adate) :
          val = val.isoformat()
        #   print colname + " is a datetime"
        if colname in event_id_tags :
          event_id[colname] = val
        elif  colname in scan_id_tags :
          scan_id[colname] = val
        elif  colname in checksum_tags :
          checksum[colname] = val
        else :
          scan[colname] = val
        
    doc = { 'type'    : 'scan',
            'event_id': event_id, 
            'scan_id' : scan_id,
            'checksum': checksum,
            'scan'    : scan }
        
    collection.insert(doc)
    collection.ensure_index('type')
    collection.ensure_index('event_id')

File: obsolete/scan/mysql_to_mongo.pl
#!/usr/bin/perl -w

#
use DBI();
use MongoDB;

#Load configuration.
#print "Reading configuration from sql_config.pl\n";
$db = $user = $host = $pass = "";

$db   = "arachne";
$host = "localhost";
$user = "tagg";
$pass = "";

$dbh = DBI->connect("DBI:mysql:database=$db;host=$host","$user", "$pass",{'RaiseError' => 1});

my $client     = MongoDB::MongoClient->new(host => 'localhost', port => 27017);
my $database   = $client->get_database( 'mydb' );
my $collection = $database->get_collection( 'tscans' );
# my $id         = $collection->insert({ some => 'data' });
# my $data       = $collection->find_one({ _id => $id });
$res = $dbh->selectall_arrayref("select * from tscans",{ Slice => {} });

for $ref (@$res) {
  $collection->insert($ref);
}
